### PR TITLE
Make periodic exporting MetricReader env vars stable

### PR DESCRIPTION
**Pull request by pellared**
## What

Mark `OTEL_METRIC_EXPORT_INTERVAL`, `OTEL_METRIC_EXPORT_TIMEOUT` environment variables as stable.

## Why

I think this change is representing the current state as all of these env vars are already implemented in OTel PHP, OTel Python, and partially by OTel Java.

See:
- [`OTEL_METRIC_EXPORT_INTERVAL` usage](https://github.com/search?p=1&q=org%3Aopen-telemetry+OTEL_METRIC_EXPORT_INTERVAL&type=Code)
- [`OTEL_METRIC_EXPORT_TIMEOUT` usage](https://github.com/search?p=1&q=org%3Aopen-telemetry+OTEL_METRIC_EXPORT_TIMEOUT&type=Code)

Created because of: https://github.com/open-telemetry/opentelemetry-dotnet/pull/3424

### CODE REVIEW

File: specification/sdk-environment-variables.md [hunk 1]
```diff
@@ -237,7 +237,7 @@ Known values for `OTEL_LOGS_EXPORTER` are:
 
 ## Metrics SDK Configuration
 
-**Status**: [Experimental](document-status.md)
+**Status**: [Stable](document-status.md)
```
Comment (reyang) [inline specification/sdk-environment-variables.md:240]: The `OTEL_METRIC_EXPORT_INTERVAL` and `OTEL_METRIC_EXPORT_TIMEOUT` for Periodic exporting MetricReader might be misleading. For traces, `OTEL_BSP_SCHEDULE_DELAY`, `OTEL_BSP_EXPORT_TIMEOUT` are used for a Batch Span Processor.

In addition, `OTEL_METRIC_EXPORT_INTERVAL` could be misleading if a configured exporter/reader is using pull mode (e.g. Prometheus) - imagine some operator trying to increase the Prometheus scraping frequency and got really puzzled by this environment variable.

**Comment by pellared** [comment]
PTAL @open-telemetry/java-instrumentation-maintainers @open-telemetry/python-maintainers @open-telemetry/php-approvers @open-telemetry/wg-prometheus

File: specification/sdk-environment-variables.md (hunk 1 above)
Comment (pellared) [reply specification/sdk-environment-variables.md:240]: Do you think the names should be changed or should we just make the description+notes more precise?

Comment (reyang) [reply specification/sdk-environment-variables.md:240]: > Do you think the names should be changed or should we just make the description+notes more precise?

I think description/notes.

**Review by reyang — CHANGES_REQUESTED** [review]
The `OTEL_METRICS_EXEMPLAR_FILTER` is still in very early stage and should remain as Experimental, consider marking the "Metrics SDK Configuration" status as Mixed, and make the "Periodic exporting MetricReader" section Stable.

In addition, consider making an update in https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md#environment-variables. Doesn't have to be in this PR though.

File: specification/sdk-environment-variables.md (hunk 1 above)
Comment (pellared) [reply specification/sdk-environment-variables.md]: @reyang did my best to address it. PTAL

**Review by reyang — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Pull request merged**